packages/frontend-shared/cypress/support/component.ts

```diff
@@ -11,13 +11,67 @@ declare global {
   namespace Cypress {
     interface Chainable {
       gqlStub: typeof GQLStubRegistry
+      /**
+       * Verify that the subject is within the bounds of the root viewport and not hidden outside a root scroll pane
+       * Note that this does not consider the condition of nested scrollpanes, so the subject could be within the root
+       * pane but hidden outside of a child pane.
+       */
+      validateWithinViewport(): Chainable<JQuery<HTMLElement>>
     }
   }
 }
 
 cy.i18n = defaultMessages
 cy.gqlStub = GQLStubRegistry
 
+function logInternal<T> (name: string | Partial<Cypress.LogConfig>, cb: (log: Cypress.Log | undefined) => Cypress.Chainable<T>, opts: Partial<Cypress.Loggable> = {}): Cypress.Chainable<T> {
+  const _log = typeof name === 'string'
+    ? Cypress.log({ name, message: '' })
+    : Cypress.log(name)
+
+  return cb(_log).then<T>((val) => {
+    _log?.end()
+
+    return val
+  })
+}
+
+function validateWithinViewport (subject: JQuery<HTMLElement>): Cypress.Chainable<JQuery<HTMLElement>> {
+  function areFullyOverlapping (elementBounds: DOMRect, scrollBounds: DOMRect) {
+    const isBoundedLeft = elementBounds.right <= scrollBounds.right
+    const isBoundedRight = elementBounds.left >= scrollBounds.left
+    const isBoundedTop = elementBounds.bottom <= scrollBounds.bottom
+    const isBoundedBottom = elementBounds.top >= scrollBounds.top
+
+    return isBoundedLeft && isBoundedRight && isBoundedTop && isBoundedBottom
+  }
+
+  function _validateWithinViewport () {
+    /* Naive implementation - does not consider nested scroll containers, only checks root container bounds */
+    cy.document({ log: false }).then((doc) => {
+      // Build rectangle reflecting bounds of root scroll container (or body if not scrollable)
+      const rootViewport = doc.scrollingElement ?? doc.body
```

Comment:
I suppose the next iteration of this command and receive a selector as a parameter for the parent scroll pane and then we could default to this behavior.